structopt-derive/src/lib.rs

```diff
@@ -282,7 +283,7 @@ fn gen_clap(attrs: &[Attribute]) -> GenOutput {
     let attrs = Attrs::from_struct(attrs, Sp::call_site(name), Sp::call_site(DEFAULT_CASING));
     let tokens = {
         let name = attrs.cased_name();
-        let methods = attrs.methods();
+        let methods = attrs.clone().top_level_methods();
```

Comment:
Why are these clone needed now?

Reply:
i just realized that this is quite complicated question, let me think...

Reply:
In my previous attempt these methods mutated `self`. Hence we happen to reuse `attrs` here first call would add methods that are acceptable for `clap::App` but not for `clap::Arg`. First we used this to generate `App` and later we generated `Arg`. My current approach is different, these methods mutate self no longer.